pkg/artifactory/resource_artifactory_webhook.go

```diff
@@ -0,0 +1,631 @@
+		anyRemote := criteria["any_remote"].(bool)
+		repoKeys := criteria["repo_keys"].(*schema.Set).List()
+
+		if (anyLocal == false && anyRemote == false) && len(repoKeys) == 0 {
+			return fmt.Errorf("repo_keys cannot be empty when both any_local and any_remote are false")
+		}
+
+		return nil
+	}
+
+	var buildCriteriaValidation = func(criteria map[string]interface{}) error {
+		log.Print("[DEBUG] buildCriteriaValidation")
+
+		anyBuild := criteria["any_build"].(bool)
+		selectedBuilds := criteria["selected_builds"].(*schema.Set).List()
+
+		if anyBuild == false && len(selectedBuilds) == 0 {
+			return fmt.Errorf("selected_builds cannot be empty when any_build is false")
+		}
+
+		return nil
+	}
+
+	var releaseBundleCriteriaValidation = func(criteria map[string]interface{}) error {
+		log.Print("[DEBUG] releaseBundleCriteriaValidation")
+
+		anyReleaseBundle := criteria["any_release_bundle"].(bool)
+		registeredReleaseBundlesNames := criteria["registered_release_bundle_names"].(*schema.Set).List()
+
+		if anyReleaseBundle == false && len(registeredReleaseBundlesNames) == 0 {
+			return fmt.Errorf("registered_release_bundle_names cannot be empty when any_release_bundle is false")
+		}
+
+		return nil
+	}
+
+	var domainCriteriaValidationLookup = map[string]func(map[string]interface{}) error{
+		"artifact":                   repoCriteriaValidation,
+		"artifact_property":          repoCriteriaValidation,
+		"docker":                     repoCriteriaValidation,
+		"build":                      buildCriteriaValidation,
+		"release_bundle":             releaseBundleCriteriaValidation,
+		"distribution":               releaseBundleCriteriaValidation,
+		"artifactory_release_bundle": releaseBundleCriteriaValidation,
+	}
+
+	var eventTypesDiff = func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+		log.Print("[DEBUG] eventTypesDiff")
+
+		eventTypes := diff.Get("event_types").(*schema.Set).List()
+		if len(eventTypes) == 0 {
+			return nil
+		}
+
+		eventTypesSupported := domainEventTypesSupported[webhookType]
+		for _, eventType := range eventTypes {
+			if !contains(eventTypesSupported, eventType.(string)) {
+				return fmt.Errorf("event_type %s not supported for domain %s", eventType, webhookType)
+			}
+		}
+		return nil
+	}
+
+	var criteriaDiff = func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+		log.Print("[DEBUG] criteriaDiff")
+
+		criteria := diff.Get("criteria").(*schema.Set).List()
+		if len(criteria) == 0 {
+			return nil
+		}
+
+		return domainCriteriaValidationLookup[webhookType](criteria[0].(map[string]interface{}))
+	}
+
+	return &schema.Resource{
+		SchemaVersion: 1,
+		CreateContext: createWebhook,
+		ReadContext:   readWebhook,
+		UpdateContext: updateWebhook,
+		DeleteContext: deleteWebhook,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema:        domainSchemaLookup[webhookType],
+		CustomizeDiff: customdiff.All(
```

Comment:
Follow other providers' example of validating cross-attributes, like from AWS: https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/ec2/network_interface.go#L196